### ISSUE TITLE
Handle the special case of bbox or geometries spanning the anti-meridian (longitude 180 degrees)

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#185]
In #96 *the special case of geographic coordinates when longitude west and east coordinates span anti-meridian* was not yet resolved.

This needs some logic at least in `GeoBox`, `ProjBox`, `BoxCoords`, and simple geometry classes too.

How to solve?
* respect GeoJSON like anti-meridian spanning bounding boxes like [177.0, -20.0, -178.0, -16.0]
* expect geometry objects to be splitted to multi-geometries as suggested by GeoJSON when a geometry spans the anti-meridian (anyway still the minimum bounding box of such a multi geometry is a single box object like in example above)
* the geobase package supports both geographic and projected coordinates/positions and geometry objects
  * so this behaviour should apply only when known to be handling geographic coordinates

**Comment by navispatial** [comment]
[RFC 7946)(https://tools.ietf.org/html/rfc7946) about [GeoJSON](https://geojson.org/) talks about this issue in chapter 5 about bounding boxes:

```
   Consider a set of point Features within the Fiji archipelago,
   straddling the antimeridian between 16 degrees S and 20 degrees S.
   The southwest corner of the box containing these Features is at 20
   degrees S and 177 degrees E, and the northwest corner is at 16
   degrees S and 178 degrees W.  The antimeridian-spanning GeoJSON
   bounding box for this FeatureCollection is

   "bbox": [177.0, -20.0, -178.0, -16.0]

   and covers 5 degrees of longitude.

   The complementary bounding box for the same latitude band, not
   crossing the antimeridian, is

   "bbox": [-178.0, -20.0, 177.0, -16.0]

   and covers 355 degrees of longitude.

   The latitude of the northeast corner is always greater than the
   latitude of the southwest corner, but bounding boxes that cross the
   antimeridian have a northeast corner longitude that is less than the
   longitude of the southwest corner.
```

**Comment by navispatial** [comment]
See also
* [Geographic Bounding Boxes](https://www.jasondavies.com/maps/bounds/) by Jason Davies
* [When “one size fits all” fails: PostGIS bounding boxes for maps](https://felt.com/blog/postgis-bounding-boxes-for-maps) by Felt
* [180th meridian](https://en.wikipedia.org/wiki/180th_meridian) by Wikipedia

**Comment by navispatial** [comment]
Implemented in [geobase 1.0.0](https://github.com/navibyte/geospatial/issues/175).
